sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobOutputStream.java

```diff
@@ -135,6 +157,7 @@ public synchronized void close() throws IOException {
         } finally {
             // if close() is called again, an exception will be thrown
             this.lastError = new IOException(Constants.STREAM_CLOSED);
+            isAlreadyClosed.set(true);
```

Comment:
This isn't required anymore as you are using `compareAndSet` that'll set this boolean to `true`. Also, since the method itself is synchronized and the `isAlreadyClose` is read/updated only inside this method, `AtomicBoolean` is not required. A regular `volatile boolean` is sufficient.